src/datasets/arrow_dataset.py

```diff
@@ -4553,6 +4589,22 @@ def to_pandas(
                 for offset in range(0, len(self), batch_size)
             )
 
+    def to_spark(self, spark_session: "Optional[pyspark.sql.SparkSession]" = None) -> "pyspark.sql.DataFrame":
+        """Convert to Spark DataFrame.
```

Comment:
Here we can either add the same warning as earlier since the full dataset goes to RAM when it's converted to pandas.

Alternatively we can send the data to the spark session by batch and use spark's `concat` on the resulting dataframes - which would allow datasets bigger than the driver's memory to be sent to the executor (and to have a nice progress bar as well). This can be done later though

Reply:
@lhoestq I didn't add the warning to `to_spark` because in this case the reason to warn is the use of our `Dataset.to_pandas` and we didn't add any warning to that other method.
- If you insist in adding a warning here in `to_spark`, then I would suggest to add the same warning to `to_pandas` for consistency.

In relation with the optimization, my original idea was to implement a simple conversion from/to Spark as a first stage. Then we can get some feedback, benchmark it in different use cases, implement optimizations,...
- See internal conversation here: https://huggingface.slack.com/archives/C012N1J633M/p1672465563591499?thread_ts=1672423801.385339&cid=C012N1J633M

Reply:
> I didn't add the warning to to_spark because in this case the reason to warn is the use of our Dataset.to_pandas and we didn't add any warning to that other method.

Pandas dataframes are known to live in RAM, while spark DataFrame are known to live in multiple workers' RAM. However here we bring the dataset in the driver node RAM first - which can be confusing and lead to errors since it's usually a small machine. That's why for this case I think it can be helpful to have a warning.